tutorials/basic-topics/sorting/merge-sort.md

```diff
@@ -11,4 +11,521 @@ keywords:
+1->10->23->null
+
+After merging we get :
+
+1->3->8->10->12->23->null.
+
+------ Hence Sorted---------------------------
+
+The two lists to be merged must be ordered respectively. We can only start to merge two lists that only have one element, then we get an ordered list that has two elements, do this again (that is "recursion").
+
+    Bottom-Up Approach 
+
+    -1-> 5->  3-> 4->   0-> null
+    \  /       \ /         /
+   -1->5->null 0->3->4->null 
+      \        /         
+-1-> 0->3-> 4-> 5->null   (sorted) 
+
+----- Finally Sorted ---------------------------------
+
+    Top Down Approach
+
+    -1-> 5->  3-> 4-> 0-> null
+         /        \ 
+-1-> 5->null      3-> 4->0->null
+    /  \          /      \  
+-1->null 5->null 3->null  4->0->null
+     \   /        \        /     \
+    -1-> 5->null  3->null 4->null 0->null
+      \             \        \   /
+                 3->null    0-> 4->null
+        \              \      /  
+                    0->3->4->null
+          \          /
+        -1->0->3->4->5->null    (sorted)
+
+```
+
+## **Approach for top-down Merge sort of Linked list**
+
+*1. Keep recursively dividing the list until there is only one node in the linked list.* 
+
+*2. Sort each sublist and merge each sorted sublist in a new array.*
+
+
+### **Steps to apply merge sort to a Linked list** 
+    
+**Note: Implementation is based on Merge Sort on an array as discussed above.** 
+
+1. Divide the linked list into two equal parts until when only one element is left
+2. To Divide, we need to  find mid in the linked list using **slow and fast pointers** method 
+3. Then merge the left and the right nodes of the linked list
+
+<Tabs>
+<TabItem value="cpp" label="C++">
+
+```cpp
+/*
+* Definition for singly-linked list.
+* struct ListNode {
+*     int val;
+*     ListNode *next;
+*     ListNode() : val(0), next(nullptr) {}
+*     ListNode(int x) : val(x), next(nullptr) {}
+*     ListNode(int x, ListNode *next) : val(x), next(next) {}
+* };
+*/
+class Solution {
+public:
+    void mergeSort(ListNode **head){
+
+        ListNode *curr = *head ; // create a current pointer
+
+        ListNode * left;
+        ListNode * right;
+        // if the linked list is null or the size is one return back the same
+        
+        if(curr ==NULL || curr ->next==NULL) return;
+        
+        //call to find the middle node between left and right
+        findMiddle(curr, &left,&right);
+        
+        // call merge_sort again to divide left half
+        mergeSort(&left);
+        
+        // call merge_sort again to divide right half
+        mergeSort(&right);
+        
+        //call to merge left and right by sorting them
+        *head=merge(left,right);
+
+    }
+
+    void findMiddle(ListNode *curr,ListNode **left, ListNode **right){
+
+        ListNode* slow = curr; // make a slow pointer
+        ListNode* fast = curr -> next; // make a fast pointer
+        
+    // then we move our fast up to it not become null
```

Comment:
indentation again .. i expect this kind of stuff can be handled well before asking for a review

Reply:
bad indentation. don't resolve if you haven't fixed